File: client/lib/features/auth/data/models/user_role.dart
/// User roles matching backend UserRole enum.
enum UserRole {
  consignor('CONSIGNOR'),
  shopOwner('SHOP_OWNER'),
  superAdmin('SUPER_ADMIN'),
  moderator('MODERATOR'),
  financeAdmin('FINANCE_ADMIN');

  final String value;
  const UserRole(this.value);

  static UserRole fromString(String value) {
    return UserRole.values.firstWhere(
      (role) => role.value == value,
      orElse: () => UserRole.consignor,
    );
  }

  String get displayName {
    switch (this) {
      case UserRole.consignor:
        return 'Penitip';
      case UserRole.shopOwner:
        return 'Pemilik Toko';
      case UserRole.superAdmin:
        return 'Super Admin';
      case UserRole.moderator:
        return 'Moderator';
      case UserRole.financeAdmin:
        return 'Finance Admin';
    }
  }

  /// Returns true if this role is an admin role (has access to admin dashboard)
  bool get isAdminRole {
    return this == UserRole.superAdmin ||
        this == UserRole.moderator ||
        this == UserRole.financeAdmin;
  }
}

File: server/src/main/java/com/ahmadramadhan/mudahtitip/auth/UserRole.java
package com.ahmadramadhan.mudahtitip.auth;

/**
 * User roles in the system.
 * SHOP_OWNER: Manages a shop, receives and sells consigned products
 * CONSIGNOR: Leaves products at shops for sale (penitip)
 * SUPER_ADMIN: Platform administrator with full access
 * MODERATOR: Content moderator for quality control
 * FINANCE_ADMIN: Financial operations manager
 */
public enum UserRole {
    SHOP_OWNER,
    CONSIGNOR,
    SUPER_ADMIN,
    MODERATOR,
    FINANCE_ADMIN
}
